src/devicecontroller/DefaultDeviceController.ts

```diff
@@ -106,6 +190,73 @@ export default class DefaultDeviceController implements DeviceControllerBasedMed
     }
   }
 
+  private isWatchingForDeviceChanges(): boolean {
+    return !!this.onDeviceChangeCallback;
+  }
+
+  private ensureWatchingDeviceChanges(): void {
+    if (this.isWatchingForDeviceChanges()) {
+      return;
+    }
+    this.onDeviceChangeCallback = () => this.handleDeviceChange();
+    this.mediaDeviceWrapper.addEventListener('devicechange', this.onDeviceChangeCallback);
+  }
+
+  /**
+   * Unsubscribe from the `devicechange` event, which allows the device controller to
+   * update its device cache.
+   */
+  private stopWatchingDeviceChanges(): void {
+    if (!this.isWatchingForDeviceChanges()) {
+      return;
+    }
+    this.mediaDeviceWrapper.removeEventListener('devicechange', this.onDeviceChangeCallback);
+    this.onDeviceChangeCallback = undefined;
+  }
+
+  private shouldObserveDeviceChanges(): boolean {
+    if (this.deviceChangeObservers.size) {
+      return true;
+    }
+
+    const hasActiveDevices =
+      (this.activeDevices['audio'] && this.activeDevices['audio'].constraints !== null) ||
+      (this.activeDevices['video'] && this.activeDevices['video'].constraints !== null) ||
+      !!this.audioOutputDeviceId;
+    return hasActiveDevices;
+  }
+
+  private watchForDeviceChangesIfNecessary(): void {
+    if (this.shouldObserveDeviceChanges()) {
+      this.ensureWatchingDeviceChanges();
+    } else {
+      this.stopWatchingDeviceChanges();
+    }
+  }
+
+  async destroy(): Promise<void> {
+    // Remove device change callbacks.
+    this.stopWatchingDeviceChanges();
+
+    // Deselect any audio input devices and throw away the streams.
+    // Discard the current video device, if there is one.
+    await this.chooseAudioInputDevice(null);
+    if (this.activeDevices['video']) {
```

Comment:
Can we call chooseVideoInputDevice(null) here? It should execute a similar code:
```
    if (device === null && kind === 'video') {
      this.lastNoVideoInputDeviceCount = this.inputDeviceCount;
      if (this.activeDevices[kind]) {
        this.releaseMediaStream(this.activeDevices[kind].stream);
        delete this.activeDevices[kind];
      }
      return;
    }
```